src/command/exit_command.rs

```diff
@@ -39,13 +39,20 @@ impl ExitCommand {
 }
 
 impl Command for ExitCommand {
-    fn execute(&mut self, _prompt: &mut Prompt) -> Result<bool, i32> {
+    fn execute(&mut self, prompt: &mut Prompt) -> Result<bool, i32> {
         let matches = self.app.get_matches_from_safe_borrow(&self.args);
         if let Err(err) = matches {
             println!("{}", err);
             return Ok(false);
         }
 
+        // Set the exit code to the previous command if it was not passed
+        // explicitly to exit.
+        if self.args.len() == 0 {
+            if let Some(c) = prompt.context.borrow().env.get("?") {
+                self.code = c.parse::<i32>().unwrap()
```

Comment:
To ensure no panics at runtime, you could change the `unwrap` into:
```rust
self.code = c.parse::<i32>().unwrap_or(1)
```
Thus, if it's not an integer it will return 1 and otherwise the integer without any panic.

However, the not-a-string probably shouldn't be a 1, so you could do this:
```rust
if let Ok(c) = c.parse::<i32>() {
  self.code = c;
}
```

Reply:
Fixed to use the latter as we don't want to default exit to 1, if a previous command was not set. e.g. startup and immediate exit command.